lib/library.rb

```diff
@@ -0,0 +1,48 @@
+class Library
+  attr_reader :name, :books, :authors
+
+  def initialize(name)
+    @name = name
+    @books = []
+    @authors = []
+  end
+
+  def add_author(author)
+    authors << author
+    books << author.books
+    books.flatten!
+  end
+
+  def publication_time_frame_for(author)
```

Comment:
This method is holding on to a lot of responsibilities - what might it look like if you could ask an author for a list of their books, sorted by publication year?